src/Misc/baseError.ts

```diff
@@ -1,46 +1,66 @@
 import { ErrorCodesType } from "./errorCodes";

```

Comment:
baseError.ts should probably be renamed to runtimeError.ts or maybe just error.ts.

Does `errorCodes.ts` still need to be a separate file? Could that just go here?

Reply:
Makes sense.